### ISSUE TITLE
Additional no match cases for EmptyCatch

### ISSUE THREAD

**Issue by ben-manes** [google/error-prone#1654]
### Description of the problem / feature request:

[EmptyCatch](https://github.com/google/error-prone/blob/master/core/src/main/java/com/google/errorprone/bugpatterns/EmptyCatch.java) will skip matching when either (1) a comment or (2) a junit test is detected. It would be nice to extend this to a few more common patterns.

### Feature requests: what underlying problem are you trying to solve with this feature?

1. If the exception is named `ignored` or `expected`. I believe this used to be part of the old Google Java style, which can be seen in Guava's JUnit tests. This would also match the behavior of PMD's [EmptyCatchBlock](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#emptycatchblock) rule.
2. Include detection of TestNG as skippable.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
// Static initializer fallback from Unsafe => MethodHandles => ThreadLocal
StripedBuffer.java:290: warning: [EmptyCatch] Caught exceptions should not be ignored
      } catch (Throwable ignored) {}
```
```
// TestNG unit test, e.g. assert NPE is thrown
CaffeineTest.java:133: warning: [EmptyCatch] Caught exceptions should not be ignored
    } catch (NullPointerException expected) {}
```

### What version of Error Prone are you using?

2.4.0

**Comment by sumitbhagwani** [comment]
Looking at the style guide : https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions

either the empty catch block should've comments with "exception" being test code where the exception name ok/expected/ignored etc are ok. 

EmptyCatch check (at HEAD/master as of July 2020) doesn't not flag JUnit test code. 

So for feature requests :
1) such code should've comments
2) I think check should be extended for TestNG.

**Comment by sumitbhagwani** [comment]
#1730 should fix this issue

**Comment by ben-manes** [comment]
Thanks!
